articles/discord-bot-node/index.md

```diff
@@ -0,0 +1,143 @@
+
+Typically, Discord bots are written in [Node.js](https://nodejs.org/en/). Node is a JavaScript interpreter that runs on the desktop. Of course, it's possible to write a bot in [other languages](https://discordapi.com/unofficial/libs.html). For some people this may be preferable. If you have lots of users, you will need an efficient language ([unlike javascript](https://benchmarksgame-team.pages.debian.net/benchmarksgame/fastest/node-gpp.html)). We will use the JavaScript library because it is the most popular. But, many of the other libraries are somewhat similar. Once you are done reading this article, try looking at some [Rust tutorials](https://www.youtube.com/watch?v=sOA6rSRCqPw&list=PLPwSz_Jcam3xVjrTAYgIHvf1Jq94yrRXp).
+<!--more-->
+## Get Started By Making a Bot
+
+### Creating The Bot
+First, you need to visit the [Discord Developer Portal](https://discordapp.com/developers/applications/). Click the "New Application" button. Give your bot a name, and click "Create".
+
+![The New Application button is in the top right corner](./discordbot1.png)
+
+![Create](/engineering-education/discord-bot-node/discordbot2.png)
+
+You'll get to your application's page, where if you want, you can give it a description and an avatar.
+
+On the left, you should see a "Bot" tab. Click on that, and then click on the button that says, "Add Bot".
+
+![The tab that says “Bot” is on the right side of the page](/engineering-education/discord-bot-node/discordbot3.png)
+
+![The “Add Bot” button is in the top right corner](/engineering-education/discord-bot-node/discordbot4.png)
+
+Give your new bot a name and an icon to make it stand out.
+
+### Adding The Bot To Your Server
+You'll need to use a link that looks something like this:
+
+```
+https://discord.com/oauth2/authorize?client_id=CLIENT_ID&scope=bot
+```
+
+You'll need to replace `CLIENT_ID` with the client ID of your bot, which you can find on the application page.
+
+![The client ID is on the “General Information” tab](/engineering-education/discord-bot-node/discordbot6.png)
+
+Once you've traveled to this page, you'll need to specify which server to add it to. It needs to be a server where you have administrative privileges. Then click "Authorize", and you'll be ready!
+
+## Set up Your Project
+You'll need to install [Node](https://nodejs.org/en/). Discord.js requires at least version 12 of Node. You can check your version by running `node -v` in a terminal or command prompt. If the version number is at least `12.0`, you're good.
+
+![The version number should be at least 12](/engineering-education/discord-bot-node/gdiscordbot7.png)
+
+Make a folder for your project. You can call it anything. Once you're in that folder, type `npm install discord.js`. It will take a bit of time. You might see a few warnings, but that's perfectly normal. Make sure you have administrative privileges if you’re on Windows, and use `sudo` on Linux.
```

Comment:
You should make it clear which shell you are using in your screenshots. Also, you should make it clear that they need to navigate to the folder they just created in the terminal and then run the command you just gave them in the terminal (beginner/windows users will probably create the folder using the UI instead of CLI). In addition, you should briefly explain npm unless your target audience should only be intermediate/advanced users...